src/Analysis/Ast/Impl/Caching/CacheFolders.cs

```diff
@@ -38,10 +38,7 @@ internal static class CacheFolders {
             return filePath;
         }
 
-        public static string GetCacheFolder(IServiceContainer services) {
-            var platform = services.GetService<IOSPlatform>();
-            var logger = services.GetService<ILogger>();
-
+        public static string GetCacheFolder(IOSPlatform platform, ILogger logger = null) {
```

Comment:
Was logging every time I called it, so wanted to have an option not to log with the optional parameter.

Reply:
Caller then should be logging. I.e. we output cache folder every time so user knows what to clear if needed.

Reply:
It was being called twice in initialization and outputting the same information so I had one call log and the other was silent.